src/driver/aurora-data-api/AuroraDataApiDriver.ts

```diff
@@ -534,6 +534,10 @@ export class AuroraDataApiDriver implements Driver {
     normalizeDefault(columnMetadata: ColumnMetadata): string | undefined {
         const defaultValue = columnMetadata.default;
 
+        if (defaultValue === null) {
+            return undefined
```

Comment:
Should we maybe be differentiating here between `null` and `undefined`? When the user sets the column default explicitly to `null` perhaps they would want the column generated to also explicitly specify `DEFAULT NULL`, even if that is probably unnecessary.

It is also already dealt with in places like this anyway:

https://github.com/typeorm/typeorm/blob/2fa6231e5903f2238b6714f52c90aaf2fe8e57eb/src/driver/mysql/MysqlQueryRunner.ts#L1877-L1878

Reply:
I've checked MySQL and Postgres databases and they are storing both `null` and `undefined` defaults like `(NULL)` value, therefore there is no difference on database level between `null` and `undefined`.

So, we can't find the difference between these values during synchronization process.

Reply:
This problem was also brought up somewhat in https://github.com/typeorm/typeorm/pull/7229#discussion_r551287724 but now we have TypeScript 4.2 with support for the `??` operator. Maybe we could start using `?? null` when comparing all the fields to avoid these situations?

Now that I think about it more it might actually make sense to have such a mapping in `normalizeDefault` but I think it should be `if (defaultValue === undefined) return null` rather than vice-versa. It's a small change but I think it might save some headaches in a few months when someone else is looking at it: it makes more sense that we are saying "if the column default wasn't defined, we default to `null`" rather than "if the column was explicitly set to `null` we change it back to not being defined (and then assume that elsewhere the code will treat `undefined` as `null`)".